client/gutenberg/extensions/mailchimp/view.js

```diff
@@ -0,0 +1,85 @@
+/**
+ * Internal dependencies
+ */
+import './view.scss';
+
+const blockClassName = 'wp-block-jetpack-mailchimp';
+
+const JetpackEmailSubscribe = {
+	validateEmail: function( email ) {
+		const re = /^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+		return re.test( String( email ).toLowerCase() );
+	},
+	fetch: ( blogId, email ) => {
+		const url =
+			'https://public-api.wordpress.com/rest/v1.1/sites/' +
+			encodeURI( blogId ) +
+			'/email_follow/subscribe?email=' +
+			encodeURI( email );
+		return new Promise( function( resolve, reject ) {
+			const xhr = new XMLHttpRequest();
+			xhr.open( 'GET', url );
+			xhr.onload = function() {
+				if ( xhr.status === 200 ) {
+					const res = JSON.parse( xhr.responseText );
+					resolve( res );
+				} else {
+					const res = JSON.parse( xhr.responseText );
```

Comment:
Would we expect an non-JSON string here?

Reply:
I don't _think_ so, but maybe a question for @artpi since that API is all him?

Reply:
I am not sure what should happen here because of the envelope
. I suggest showing a generic error